hypothesis-python/src/hypothesis/strategies/_internal/core.py

```diff
@@ -1463,29 +1467,32 @@ def composite(f: Callable[..., Ex]) -> Callable[..., SearchStrategy[Ex]]:
     else:
         special_method = None
 
-    argspec = getfullargspec(f)
+    sig = signature(f)
```

Comment:
To preserve behavior of `getfullargspec`:

`sig = signature(f, follow_wrapped=True)`